stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/repository/ReactionRepositoryTest.kt

```diff
@@ -1,86 +1,102 @@
+    private lateinit var reactionDao: ReactionDao
+    private lateinit var currentUser: User
+    private lateinit var reactionRepo: ReactionRepository
 
     @Before
-    fun beforeEach() {
+    fun setup() {
         runBlocking {
-            helper.insertMessage(data.message1)
+            currentUser = randomUser()
+            reactionDao = mock()
+            reactionRepo = ReactionRepositoryImpl(reactionDao) { currentUser }
         }
     }
 
     @Test
-    fun testInsertAndRead() = runBlocking {
-        helper.insertReaction(data.reaction1)
-        val reaction = helper.selectUserReactionsToMessage(data.reaction1.messageId, data.reaction1.user!!.id).first()
-        Truth.assertThat(reaction).isEqualTo(data.reaction1)
+    fun `Reaction should be stored in Db`() = runBlockingTest {
+        val messgId = randomString(10)
+        val reaction = randomReaction(messageId = messgId, user = currentUser, type = "love")
+
+        reactionRepo.insertReaction(reaction)
+
+        verify(reactionDao).insert(
+            argThat {
+                this.messageId == messageId && this.userId == currentUser.id
+            }
+        )
     }
 
     @Test
-    fun testSyncNeeded() = runBlocking {
-        data.reaction1.syncStatus = SyncStatus.FAILED_PERMANENTLY
-        val reaction2 =
-            data.reaction1.copy().apply { type = "love"; syncStatus = SyncStatus.SYNC_NEEDED }
-        helper.insertReaction(data.reaction1)
-        helper.insertReaction(reaction2)
-        var reactions = helper.selectReactionsSyncNeeded()
-        Truth.assertThat(reactions.size).isEqualTo(1)
-        Truth.assertThat(reactions.first().syncStatus).isEqualTo(SyncStatus.SYNC_NEEDED)
-
-        reactions = chatDomainImpl.retryReactions()
-        Truth.assertThat(reactions.size).isEqualTo(1)
-        Truth.assertThat(reactions.first().syncStatus).isEqualTo(SyncStatus.COMPLETED)
-
-        reactions = helper.selectReactionsSyncNeeded()
-        Truth.assertThat(reactions.size).isEqualTo(0)
+    fun `Shouldn't allow storing reaction with empty messageId`() = runBlockingTest {
+        val reaction = Reaction(messageId = "", user = currentUser, type = "love")
+
+        assertThrows<IllegalArgumentException> { reactionRepo.insertReaction(reaction) }
+
+        verifyZeroInteractions(reactionDao)
     }
 
     @Test
-    fun testUpdate() = runBlocking {
-        val reaction1Updated =
-            data.reaction1.copy().apply { extraData = mutableMapOf("theanswer" to 42.0) }
-        helper.insertReaction(data.reaction1)
-        helper.insertReaction(reaction1Updated)
-
-        val reaction = helper.selectUserReactionsToMessage(data.reaction1.messageId, data.reaction1.user!!.id).first()
-        Truth.assertThat(reaction).isEqualTo(reaction1Updated)
+    fun `Shouldn't allow storing reaction with empty type`() = runBlockingTest {
+        val reaction = Reaction(messageId = randomString(10), user = currentUser, type = "")
```

Comment:
Again. It's better to use factory method, constructor can be changed, it can require additional args. If it happens it's easier to change only factory method